staging/src/k8s.io/api/core/v1/register.go

```diff
@@ -90,6 +90,13 @@ func addKnownTypes(scheme *runtime.Scheme) error {
 		&ConfigMapList{},
 	)
 
+	// Allow them to be decoded across all version in this scheme
+	scheme.AddUnversionedTypes(SchemeGroupVersion,
+		&Pod{},
+		&ReplicationController{},
+		&Service{},
+	)
```

Comment:
while this might be a fix for the related issue, i'm not positive we want to do that in the wider picture.
my initial impression is - no.
but i will defer to api-machinery

/sig api-machinery

Reply:
Understand.

BTW, I saw this approach "AddUnversionedTypes" which has been used in api-machinery.

Reply:
this is not correct... these are versioned types. Unversioned types refer to things like metav1 types, {create,get,list,delete}options, etc